src/document.jl

```diff
@@ -90,6 +90,13 @@ function parse_file(filename::AbstractString)
     XMLDocument(p)
 end
 
+function parse_file(filename::AbstractString, encoding, options::Integer)
+    p = ccall((:xmlReadFile,libxml2), Xptr, (Cstring, Ptr{Cchar}, Cint),
```

Comment:
Just `encoding` strings with embedded nul's. I'd rather have this API be closer to the extensively documented upstream C API, even if that means a few methods are a little unsafe. There are bigger issues to worry about here, like memory management being kind of a mess without reference counts.